File: src/main/java/com/lannstark/lec07/Lec07Main.java
package com.lannstark.lec07;

import java.io.File;
import java.io.IOException;

public class Lec07Main {

  public static void main(String[] args) throws IOException {
      JavaFilePrinter javaFilePrinter = new JavaFilePrinter();

      File file = new File(".");
      javaFilePrinter.readFile(file.getAbsolutePath() + "/a.txt");
  }

}

File: src/main/java/com/lannstark/lec07/Lec07Main.kt
package com.lannstark.lec07

fun main() {
    FilePrinter().readFile()
}

// try catch 구문은 Java와 문법적으로 동일하다 (finally도 동일)
fun parseIntOrThrow(str: String): Int {
    try {
        return str.toInt()
    } catch (e: NumberFormatException) {
        throw IllegalArgumentException("주어진 ${str}는 숫자가 아닙니다")
    }
}

// Kotlin에서는 try catch 구문 역시 expression이다
fun pareseIntOrThrowV2(str: String): Int? {
    return try {
        str.toInt()
    } catch (e: NumberFormatException) {
        null
    }
}


